conans/test/unittests/client/graph/build_mode_test.py

```diff
@@ -0,0 +1,73 @@
+import unittest
+
+from conans.client.graph.build_mode import BuildMode
+from conans.model.ref import ConanFileReference
+from conans.errors import ConanException
+
+from conans.test.utils.tools import TestBufferConanOutput
+from conans.test.utils.conanfile import MockConanfile
+
+
+class BuildModeTest(unittest.TestCase):
+
+    def setUp(self):
+        self.output = TestBufferConanOutput()
+        self.conanfile = MockConanfile(None)
+
+    def test_valid_params(self):
+        build_mode = BuildMode(["outdated", "missing"], self.output)
+        self.assertTrue(build_mode.outdated)
+        self.assertTrue(build_mode.missing)
+        self.assertFalse(build_mode.never)
+
+        build_mode = BuildMode(["never"], self.output)
+        self.assertFalse(build_mode.outdated)
+        self.assertFalse(build_mode.missing)
+        self.assertTrue(build_mode.never)
+
+    def test_invalid_configuration(self):
+        with self.assertRaises(ConanException):
```

Comment:
I would test the message exception too, it will help me to know why it is failing here. Thanks